### PR TITLE
upgrade libgit2 to v1.3.0

### PR DESCRIPTION
**Pull request by medo-freiheit**
upgrade alpine to alpine:3.15
upgrade git2go to v33 to support libgit2 v1.3.0 [link](https://github.com/libgit2/git2go#which-go-version-to-use)

### CODE REVIEW

File: build/CMakeCache.txt [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# This is the CMakeCache file.
```
Comment (sven-urbanski-freiheit-com) [inline build/CMakeCache.txt:1]: please don't commit the `build` directory.
And please add it to the `.gitignore` file so that it doesn't happen accidentally.

File: go.sum [hunk 2]
```diff
@@ -270,6 +270,8 @@ github.com/leodido/go-urn v1.2.0 h1:hpXL4XnriNwQ/ABnpepYM/1vCLWNDfUNts8dX3xTG6Y=
 github.com/leodido/go-urn v1.2.0/go.mod h1:+8+nEpDfqqsY+g338gtMEUOtuK+4dEMhiQEgxpxOKII=
 github.com/libgit2/git2go/v31 v31.4.14 h1:6GOd3965D9e/+gjxCwZF4eQ+vB9kKB4yKFqdQr6XZ2E=
 github.com/libgit2/git2go/v31 v31.4.14/go.mod h1:c/rkJcBcUFx6wHaT++UwNpKvIsmPNqCeQ/vzO4DrEec=
+github.com/libgit2/git2go/v33 v33.0.4 h1:37xovFBzibhDEdQRLbfWwx3a44JhOIY06UICn2teenc=
+github.com/libgit2/git2go/v33 v33.0.4/go.mod h1:KdpqkU+6+++4oHna/MIOgx4GCQ92IPCdpVRMRI80J+4=
```
Comment (sven-urbanski-freiheit-com) [inline go.sum:274]: please run `go mod tidy`, it should get rid of the old version 31

File: pkg/api/node_modules/.yarn-integrity [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (sven-urbanski-freiheit-com) [inline pkg/api/node_modules/.yarn-integrity:1]: I don't think we need this file, please remove.

File: pkg/api/yarn.lock [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (sven-urbanski-freiheit-com) [inline pkg/api/yarn.lock:1]: I don't think we need this file, please remove.

**Review by sven-urbanski-freiheit-com — CHANGES_REQUESTED** [review]
please see my other comments

**Review by hannesg — APPROVED** [review]

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

File: services/cd-service/Dockerfile [hunk 5]
```diff
@@ -1,4 +1,4 @@
-FROM alpine:3.14
+FROM alpine:3.15
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/Dockerfile:1]: please also change docker/build/Dockerfile

Comment (medo-freiheit) [reply services/cd-service/Dockerfile:1]: I'm kinda trying to create a new PR because I didn’t mange to get ride of the 
Codereviewr Check error

Comment (sven-urbanski-freiheit-com) [reply services/cd-service/Dockerfile:1]: that's fine.
Then I would suggest this: close this PR and add a comment here that links to the new PR

File: docker/build/Dockerfile [hunk 6]
```diff
@@ -1,4 +1,4 @@
-FROM alpine:3.14
+FROM alpine:3.15
 LABEL org.opencontainers.image.source https://github.com/freiheit-com/kuberpult
 RUN apk add --no-cache libgit2 libgit2-dev go protoc make pkgconfig git
```
Comment (sven-urbanski-freiheit-com) [inline docker/build/Dockerfile:3]: please remove all .DS_Store files from the PR (and add them to gitignore)

File: services/cd-service/repository_remote/HEAD [hunk 7]
```diff
@@ -0,0 +1 @@
+ref: refs/heads/master
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/repository_remote/HEAD:1]: do not commit repository_remote!

File: services/frontend-service/package.json [hunk 8]
```diff
@@ -54,5 +54,5 @@
       "last 1 safari version"
     ]
   },
-  "proxy": "http://localhost:8081"
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/package.json:57]: is this change necessary?

Comment (medo-freiheit) [reply services/frontend-service/package.json]: No, resolved

**Comment by medo-freiheit** [comment]
Please check the latest commit

File: docker/build/Dockerfile [hunk 9]
```diff
@@ -1,4 +1,4 @@
...
-RUN wget https://github.com/argoproj/argo-cd/releases/download/v2.1.2/argocd-linux-amd64 -O /usr/local/bin/argocd && chmod +x /usr/local/bin/argocd
+#RUN wget https://github.com/argoproj/argo-cd/releases/download/v2.1.2/argocd-linux-amd64 -O /usr/local/bin/argocd && chmod +x /usr/local/bin/argocd
```
Comment (sven-urbanski-freiheit-com) [inline docker/build/Dockerfile:4]: this should not be commented out

File: .gitignore [hunk 10]
```diff
@@ -1,6 +1,9 @@
 bin
 services/cd-service/repository_checkedout
+services/cd-service/repository_remote
 services/cd-service/repository
+services/cd-service/.DS_Store
```
Comment (sven-urbanski-freiheit-com) [inline .gitignore:5]: please change this to `.DS_Store` as those files can appear anywhere

File: services/cd-service/Makefile [hunk 11]
```diff
@@ -50,10 +50,10 @@ ifeq ($(COMPILE_WITH_DOCKER),true)
 # note repository must be named repository otherwise kuberpult cannot see the repository
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/Makefile:50]: there are still DS_Store files in this PR, please remove all of them
